contracts/mainnet/connectors/basic-ERC4626/main.sol

```diff
@@ -0,0 +1,155 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+/**
+ * @title Basic D.
+ * @dev Deposit, Mint, Withdraw, & Redeem from ERC4626 DSA.
+ */
+
+// import { IERC4626 } from "@openzeppelin/contracts-latest/interfaces/IERC4626.sol";
+import { IERC4626 } from "./interface.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+import { Events } from "./events.sol";
+
+abstract contract BasicConnector is Events, DSMath, Basic {
+	/**
+	 * @dev Deposit underlying asset to ERC4626 Vault.
+	 * @notice Mints vault shares by depositing exactly amount of underlying assets
+	 * @param token ERC4626 Token address.
+	 * @param amt The amount of the token to deposit. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+
+	function deposit(
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	) public returns (string memory _eventName, bytes memory _eventParam) {
+		uint256 _amt = getUint(getId, amt);
+		IERC4626 vaultTokenContract = IERC4626(token);
+
+		address _underlyingToken = vaultTokenContract.asset();
+		TokenInterface underlyingTokenContract = TokenInterface(_underlyingToken);
+
+		_amt = _amt == uint256(-1) ? underlyingTokenContract.balanceOf(address(this)) : _amt;
+
+		approve(underlyingTokenContract, token, _amt);
+
+		vaultTokenContract.deposit(_amt, address(this));
+		setUint(setId, _amt);

```

Comment:
Get final balance of vault token.
require(minShares < (final balance - initial balance))